File: solutions/scala/etl/1/src/main/scala/Etl.scala
object Etl {
  def transform(scoreMap: Map[Int, Seq[String]]): Map[String, Int] = {
    var result = Map[String, Int]()
    for ((key, seq) <- scoreMap) {
      for (s <- seq) {
        result += (s.toLowerCase -> key)
      }
    }
    return result
  }
}

File: solutions/typescript/etl/1/etl.ts
export function transform(dict: Map<string, string[]>): {} {

  const result: {[key: string]: number} = {}
  for (let [key, values] of Object.entries(dict)) {
    for (let value of values) {
      result[value.toLowerCase()] = Number(key)
    }
  }

  return result
}